### PR TITLE
Fix backup and restore in presets by moving expert mode check

### PR DESCRIPTION
**Pull request by haslinghuis**
After merging https://github.com/betaflight/betaflight-configurator/pull/2866 the backup and restore buttons did not work.

Moving the check from closing the connection to opening a new connection should solve it.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by mrRobot62** [comment]
made a couple of different tests and compares. Backup as well as loading works as designed.

Did this tests on Mac 10.15.7 (Catalina)
Good job

**Review by limonspb — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by blckmn — APPROVED** [review]

**Comment by limonspb** [comment]
just tested. Works as expected. Lets merge

**Pull request merged**
